xbmc/filesystem/ZipFile.cpp

```diff
@@ -420,12 +427,14 @@ int CZipFile::UnpackFromMemory(std::string& strDest, const std::string& strInput
             break;  // header found
           i++;
         }
-        mZipItem.crc32 = strInput[i + 4] || strInput[i + 5] << 8 || strInput[ i+ 6] << 16 || 
-          strInput[i + 7] << 24;   // ZIP is little endian
-        mZipItem.csize = strInput[i + 8] || strInput[i + 9] << 8 || strInput[i + 10] << 16 || 
-          strInput[i + 11] << 24;
-        mZipItem.usize = strInput[i + 12] || strInput[i + 13] << 8 || strInput[i + 14] << 16 || 
-          strInput[i + 15] << 24;
+        // ZIP is little endian:
+        mZipItem.crc32 = (uint8_t)strInput[i + 4] | (uint8_t)strInput[i + 5] << 8 |
```

Comment:
Please use static_cast<uint8_t>(....) - this is one of our code guideline conventions

Reply:
Ok did it. I think my commit already got pushed into this PR, can you confirm?